### PR TITLE
CASMINST-5067: Implementation code for NCN 1.3

### PR DESCRIPTION
**Pull request by mantoanibroadwing**
Goss for https://github.com/Cray-HPE/docs-csm/blob/main/operations/validate_csm_health.md#13-check-of-system-management-monitoring-tools

### CODE REVIEW

File: goss-testing/tests/ncn/goss-check-system-management-tools.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    exec: "kubectl -n services get svc cray-dns-powerdns-cmn-udp"
+  switch_ping-status:
+    exit-status: 0
+    exec: "canu test --csm 1.2 --json --username admin --password ${SW_ADMIN_PASSWORD} |grep ping"
+    stdout:
+      - "PASS"
+    timeout: 60000
+  bgp_established_mellanox:
+    exit-status: 0
+    exec: "sshpass -p ${SW_ADMIN_PASSWORD} ssh admin@sw-spine-001.hmn cli \"enable\" \"show ip bgp vrf all summary\""
```
Comment (mharding-hpe) [inline goss-testing/tests/ncn/goss-check-system-management-tools.yaml:27]: sshpass is not installed on the NCNs currently. Is this being added?

File: goss-testing/tests/ncn/goss-check-system-management-tools.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    stdout:
+      - "ESTABLISHED"
+    timeout: 30000
+  bgp_established_aruba:
+    exit-status: 0
+    exec: "sshpass -p ${SW_ADMIN_PASSWORD} ssh admin@sw-spine-001.hmn cli \"enable\" \"show bgp all-vrf all summary\""
```
Comment (mharding-hpe) [inline goss-testing/tests/ncn/goss-check-system-management-tools.yaml:33]: Same question about sshpass here

File: goss-testing/tests/ncn/goss-check-system-management-tools.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    stdout:
+      - "Established"
+    timeout: 30000
+  k8s_service_no_ip:
+    exit-status: 1
+    exec: "kubectl get service -A | grep Load|grep -i pending"
```
Comment (mharding-hpe) [inline goss-testing/tests/ncn/goss-check-system-management-tools.yaml:39]: This subtest should use the logrun tool to capture its output. See the other Goss test files in the repo for examples of how to do this.

File: goss-testing/tests/ncn/goss-check-system-management-tools.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  k8s_MetalLB_configmap:
+    exit-status: 0
+    exec: "kubectl -n istio-system describe service istio-ingressgateway-cmn"
```
Comment (mharding-hpe) [inline goss-testing/tests/ncn/goss-check-system-management-tools.yaml:42]: This subtest should use the logrun tool to capture its output. See the other Goss test files in the repo for examples of how to do this.

File: goss-testing/tests/ncn/goss-check-system-management-tools.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    stdout:
+      - "Annotations"
+  k8s_customer_management_address:
+    exit-status: 0
+    exec: "kubectl -n metallb-system get cm metallb -o yaml"
```
Comment (mharding-hpe) [inline goss-testing/tests/ncn/goss-check-system-management-tools.yaml:47]: This subtest should use the logrun tool to capture its output. See the other Goss test files in the repo for examples of how to do this.

File: goss-testing/tests/ncn/goss-check-system-management-tools.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  check_if_prometheus_module_enabled:
+    exit-status: 0
+    exec: "ceph mgr module ls | jq '.enabled_modules'"
```
Comment (mharding-hpe) [inline goss-testing/tests/ncn/goss-check-system-management-tools.yaml:50]: This subtest should use the logrun tool to capture its output. See the other Goss test files in the repo for examples of how to do this.

**Review by mharding-hpe — CHANGES_REQUESTED** [review]

**Comment by mharding-hpe** [comment]
Also, the description for the PR should include the associated Jira ticket number

File: goss-testing/tests/ncn/goss-check-system-management-tools.yaml (hunk 1 above)
Comment (mantoanibroadwing) [reply goss-testing/tests/ncn/goss-check-system-management-tools.yaml]: Yes its being handled by a bash script that calls this goss

**Comment by rustydb** [comment]
> Also, the description for the PR should include the associated Jira ticket number

No idea which JIRA this is for based on the commit, pr description, or pr title.

File: goss-testing/tests/ncn/goss-check-system-management-tools.yaml [hunk 7]
```diff
@@ -0,0 +1,52 @@
+command:
```
[the remark this answers is not in the document]
Comment (mantoanibroadwing) [reply goss-testing/tests/ncn/goss-check-system-management-tools.yaml]: Updated on new files

File: goss-testing/tests/ncn/goss-check-system-management-tools.yaml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  assigned_ip-status:
+    exit-status: 0
+    exec: "ip addr show bond0.cmn0|grep -i inet"
+  default_gw:
+    exit-status: 0
+    exec: "ip route | grep default"
```
[the remark this answers is not in the document]
Comment (mantoanibroadwing) [reply goss-testing/tests/ncn/goss-check-system-management-tools.yaml]: Created and updated in separated file

[the remark this answers is not in the document]
Comment (mantoanibroadwing) [reply goss-testing/tests/ncn/goss-check-system-management-tools.yaml]: Created separated files and user log_run

File: goss-testing/tests/ncn/goss-check-system-management-tools.yaml (hunk 2 above)
Comment (mantoanibroadwing) [reply goss-testing/tests/ncn/goss-check-system-management-tools.yaml]: Created separated files and user log_run

File: goss-testing/tests/ncn/goss-check-system-management-tools.yaml (hunk 3 above)
Comment (mantoanibroadwing) [reply goss-testing/tests/ncn/goss-check-system-management-tools.yaml]: Created separated files and user log_run

File: goss-testing/tests/ncn/goss-check-system-management-tools.yaml (hunk 6 above)
Comment (mantoanibroadwing) [reply goss-testing/tests/ncn/goss-check-system-management-tools.yaml]: Created separated files and user log_run

**Comment by mantoanibroadwing** [comment]
> > Also, the description for the PR should include the associated Jira ticket number
> 
> No idea which JIRA this is for based on the commit, pr description, or pr title.

Added Jira ticket on title

File: goss-testing/tests/ncn/goss-k8s-metallb_configmap-status.yaml [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+{{ $scripts := .Env.GOSS_BASE | printf "%s/scripts" }}
+{{ $logrun := $scripts | printf "%s/log_run.sh" }}
+command:
+    {{ $testlabel := "check_k8s-metallb-status" }}
+    {{$testlabel}}:
+        title: Check if Metal Load Balancer for Kubernetes is running on service ingress
```
Comment (mharding-hpe) [inline goss-testing/tests/ncn/goss-k8s-metallb_configmap-status.yaml:30]:
```suggestion
        title: Verify that Metal Load Balancer for Kubernetes is running on service ingress
```

**Review by mharding-hpe — CHANGES_REQUESTED** [review]
Also, please update your PR template to include a description of where and how you tested these new tests, to ensure that they all work as expected.

**Comment by mantoanibroadwing** [comment]
Closing this PR as https://github.com/Cray-HPE/csm-testing/pull/365 was created and handled all fixes

**Pull request closed without merging**